### PR TITLE
(SIMP-9130) TPM2 Updates

### PR DESCRIPTION
**Pull request by jeannegreulich**
- Updates in the TPM2 tools caused errors in TPM2 fact
      and ability to set ownership.
    - Updated the tpm2 fact to determine the version of the tpm tools
      installed and get data appropriately.
    - Created type for new tool tpm2_changeauth which replaces
      tpm2_takeownership tool.
    - Updated manifest to call the appropriate type to
      change ownership
    
    SIMP-9130 #close

### CODE REVIEW

File: lib/facter/tpm2/util.rb [hunk 1]
```diff
@@ -21,14 +21,21 @@ module Facter::TPM2; end
...
-  def exec(cmd)
-    Facter.debug "executing '#{File.join(@prefix, cmd)}'"
-    Facter::Core::Execution.execute(File.join(@prefix, cmd))
+    cmd  = Facter::Core::Execution.which('tpm2_getcap')
+
+    # Between versions the options for 'tpm2_getcap' changed. Determine the
+    #  version and set the options.
+    output = Facter::Core::Execution.execute(%(#{cmd} -v))
+    result = output.match(/version="(\d+\.\d+\.\d+)/)
+    @version = $1
```
Comment (lnemsick-simp) [inline lib/facter/tpm2/util.rb:30]: should handle the case when result=nil

File: lib/facter/tpm2/util.rb [hunk 2]
```diff
@@ -21,14 +21,21 @@ module Facter::TPM2; end
...
   def initialize
-    @prefix = Facter::TPM2::Util.tpm2_tools_prefix
-  end
-
-  # Facter executes a CLI command using the tpm2-tools path
-  # @param [String] cmd The CLI command string for Facter to execute
-  def exec(cmd)
-    Facter.debug "executing '#{File.join(@prefix, cmd)}'"
-    Facter::Core::Execution.execute(File.join(@prefix, cmd))
+    cmd  = Facter::Core::Execution.which('tpm2_getcap')
```
Comment (lnemsick-simp) [inline lib/facter/tpm2/util.rb:24]: what if cmd returns nil?

Comment (trevor-vaughan) [reply lib/facter/tpm2/util.rb:24]: It would never be called because of the `confine` in the top-level fact.

**Comment by trevor-vaughan** [comment]
Working on the minor requested updates

**Comment by trevor-vaughan** [comment]
Rebased and updated to address review comments

**Review by trevor-vaughan — APPROVED** [review]

**Pull request merged**